services/kafka/sasl.go

```diff
@@ -102,10 +193,26 @@ func SASLVersion(kafkaVersion sarama.KafkaVersion, saslVersion *int) (int16, err
 func gssapiAuthType(authType string) int {
 	switch authType {
 	case "KRB5_USER_AUTH":
-		return sarama.KRB5_USER_AUTH
+		return kafka.KRB5_USER_AUTH
 	case "KRB5_KEYTAB_AUTH":
-		return sarama.KRB5_KEYTAB_AUTH
+		return kafka.KRB5_KEYTAB_AUTH
 	default:
 		return 0
 	}
 }
+
+// azureAD returns a new oauth2.Endpoint for the given tenant at Azure Active Directory.
+// If tenant is empty, it uses the tenant called `common`.
+//
+// For more information see:
+// https://docs.microsoft.com/en-us/azure/active-directory/develop/active-directory-v2-protocols#endpoints
+func azureAD(tenant string) oauth2.Endpoint {
+	if tenant == "" {
+		tenant = "common"
+	}
+	return oauth2.Endpoint{
+		AuthURL:       "https://login.microsoftonline.com/" + tenant + "/oauth2/v2.0/authorize",
+		TokenURL:      "https://login.microsoftonline.com/" + tenant + "/oauth2/v2.0/token",
+		DeviceAuthURL: "https://login.microsoftonline.com/" + tenant + "/oauth2/v2.0/devicecode",
+	}
+}
```

Comment:
Instead of this hand-crafted implementation, why not rely on https://pkg.go.dev/golang.org/x/oauth2@v0.23.0/endpoints?

Reply:
Yes, this was mistakenly copied.